src/coreclr/jit/fgdiagnostic.cpp

```diff
@@ -3582,6 +3583,574 @@ void Compiler::fgDebugCheckNodesUniqueness()
+                                                                       fieldStoreSize);
+                    }
+                    else
+                    {
+                        isFieldUse = true;
+                    }
+
+                    if (isFieldUse)
+                    {
+                        unsigned const fieldUseSsaNum = fieldVarDsc->GetPerSsaData(fieldSsaNum)->GetUseDefSsaNum();
+                        ProcessUse(lclNode, fieldLclNum, fieldUseSsaNum, fieldVarDsc);
+                    }
+                }
+            }
+        }
+        else
+        {
+            unsigned const ssaNum = lclNode->GetSsaNum();
+            ProcessDef(lclNode, lclNum, ssaNum, varDsc);
+
+            if (isUse)
+            {
+                unsigned useSsaNum = SsaConfig::RESERVED_SSA_NUM;
+                if (ssaNum != SsaConfig::RESERVED_SSA_NUM)
+                {
+                    useSsaNum = varDsc->GetPerSsaData(ssaNum)->GetUseDefSsaNum();
+                }
+                ProcessUse(lclNode, lclNum, useSsaNum, varDsc);
+            }
+        }
+    }
+
+    void ProcessUse(GenTreeLclVarCommon* tree, unsigned lclNum, unsigned ssaNum, LclVarDsc* varDsc)
+    {
+        // If the var is not in ssa, the tree should not have an ssa num.
+        //
+        if (!varDsc->lvInSsa)
+        {
+            if (ssaNum != SsaConfig::RESERVED_SSA_NUM)
+            {
+                SetHasErrors();
+                JITDUMP("[error] Unexpected SSA number on [%06u] (V%02u)\n", m_compiler->dspTreeID(tree), lclNum);
+            }
+            return;
+        }
+
+        // All uses of ssa vars should have valid ssa nums, unless there are no defs.
+        //
+        if (ssaNum == SsaConfig::RESERVED_SSA_NUM)
+        {
+            if (varDsc->lvPerSsaData.GetCount() > 0)
+            {
+                SetHasErrors();
+                JITDUMP("[error] Missing SSA number on use [%06u] (V%02u)\n", m_compiler->dspTreeID(tree), lclNum);
+            }
+            return;
+        }
+
+        JITDUMP("Use of V%02u.%u @ [%06u] " FMT_BB "\n", lclNum, ssaNum, m_compiler->dspTreeID(tree), m_block->bbNum);
+
+        SsaKey   key(lclNum, ssaNum);
+        SsaInfo* ssaInfo = nullptr;
+
+        if (!m_infoMap.Lookup(key, &ssaInfo))
+        {
+            ssaInfo = new (m_compiler->getAllocator(CMK_DebugOnly)) SsaInfo;
+            m_infoMap.Set(key, ssaInfo);
+        }
+
+        if (tree->OperIs(GT_PHI_ARG))
+        {
+            ssaInfo->AddPhiUse(m_block, key);
+        }
+        else
+        {
+            ssaInfo->AddUse(m_block, key);
+        }
+    }
+
+    void ProcessUses(GenTreeLclVarCommon* tree)
+    {
+        unsigned const   lclNum = tree->GetLclNum();
+        unsigned const   ssaNum = tree->GetSsaNum();
+        LclVarDsc* const varDsc = m_compiler->lvaGetDesc(lclNum);
+
+        // Handle multiple uses
+        //
+        if (tree->HasCompositeSsaName())
```

Comment:
Naturally, this will always be `false`... It would also need some tweaks if we do start renaming SSA uses, in particular, it would need to deduce which fields are actually references by the use (otherwise the "field local is in SSA => has SSA num" assertion won't be true). Delete for now?

(Or is something else planned for this?)